homeassistant/config_entries.py

```diff
@@ -750,30 +750,54 @@ def async_update_entry(
         data: dict = _UNDEF,
         options: dict = _UNDEF,
         system_options: dict = _UNDEF,
-    ) -> None:
-        """Update a config entry."""
+    ) -> bool:
+        """Update a config entry.
+
+        If the entry was changed, the update_listeners are
+        fired and this function returns True
+
+        If the entry was not changed, the update_listeners are
+        not fired and this function returns False
+        """
+        changed = False
+
         if unique_id is not _UNDEF:
+            if entry.unique_id != unique_id:
+                changed = True
             entry.unique_id = cast(Optional[str], unique_id)
 
         if title is not _UNDEF:
+            if entry.title != title:
+                changed = True
             entry.title = cast(str, title)
 
         if data is not _UNDEF:
+            if dict(entry.data) != dict(data):
+                changed = True
             entry.data = MappingProxyType(data)
 
         if options is not _UNDEF:
+            if dict(entry.options) != dict(options):
```

Comment:
Isn't it better to include conditipm in above check? No need to write the data to the entry either.

Reply:
Will adjust

Reply:
done